### ISSUE TITLE
Verify interpolation of image processors

### ISSUE THREAD

**Issue by NielsRogge** [huggingface/transformers#28180]
### Feature request

As pointed out in https://github.com/huggingface/transformers/pull/27742, some image processors might need a correction on the default interpolation method being used (resampling in Pillow). We could check this on a per-model basis.

### Motivation

Interpolation methods have a slight (often minimal) impact on performance. However it would be great to verify this on a per-model basis.

e.g. [ViT](https://github.com/huggingface/transformers/blob/main/src/transformers/models/vit/image_processing_vit.py#L52)'s image processor defaults to BILINEAR but should use BICUBIC as seen [here](https://github.com/huggingface/pytorch-image-models/blob/main/timm/models/vision_transformer.py#L1062). We can update the default values of the image processors, but can't update the configs on the hub as this would break people's fine-tuned models.

### Your contribution

I could work on this, but this seems like a good first issue for first contributors.

**Comment by amyeroberts** [comment]
@NielsRogge Thanks for opening the issue! 

It's fine to open up to the community but you'll need to add a checklist of the image processors so it's clear who is working on what and what's done as well as ideally some instructions on what it means for each one to be "done" e.g. making sure to run slow tests for models.

**Comment by nileshkokane01** [comment]
@NielsRogge ,

If I understand it correctly, we need to match the interpolation:

For example for convnext: [convnext](https://github.com/huggingface/transformers/blob/main/src/transformers/models/convnext/image_processing_convnext.py#L93) should be changed to Bicubic as per [timm/convnext](https://github.com/huggingface/pytorch-image-models/blob/main/timm/models/convnext.py#L498) .

If I understand it correctly, I can take this up for all the models. Let me know.

**Comment by NielsRogge** [comment]
Yes that is correct, see also the [original implementation](https://github.com/facebookresearch/ConvNeXt/blob/048efcea897d999aed302f2639b6270aedf8d4c8/main.py#L107). Thanks for spotting that. Hence feel free to open a PR to update this, along with the image processor created in the [conversion script](https://github.com/huggingface/transformers/blob/main/src/transformers/models/convnext/convert_convnext_to_pytorch.py#L149-L150). Ideally we assert the pixel values created by it against the original implementation, like done [here](https://github.com/huggingface/transformers/blob/main/src/transformers/models/dinov2/convert_dinov2_to_hf.py#L197-L220) for DINOv2.

**Comment by nileshkokane01** [comment]
Sure! thanks for the pointers, will work on it.

**Comment by nileshkokane01** [comment]
DieT and DPT default interpolation types matches with the original implementation default types to BICUBIC . That's what I see, let me know  if  I overlooked.

**Comment by nileshkokane01** [comment]
@NielsRogge ,

would you have a look ?

**Comment by amyeroberts** [comment]
@NielsRogge Can you please complete the checklist here?

**Comment by hritikranjan1** [comment]
@NielsRogge 
I would like to take this issue up.

**Comment by shubham-61969** [comment]
@NielsRogge , @nileshkokane01  Can I work on this issue and help in completing the checklist ?

**Comment by farrosalferro** [comment]
@NielsRogge , is it possible for me to contribute ? Thanks.

**Comment by 05kashyap** [comment]
@NielsRogge , can I work on this issue?

**Comment by NielsRogge** [comment]
Please don't ask to work on it, just open a PR for one of the models which are still open/which doesn't have a merged PR yet :)

Thanks.

**Comment by Arpit2744** [comment]
#43017 Fix default interpolation for PVT to match original implementation [(Bicubic)] @NielsRogge

**Comment by sindhukodati** [comment]
Hi @NielsRogge,

I’d like to work on fixing the default interpolation for the SegFormer
image processor by verifying it against the original implementation
and adding a test similar to existing vision models.

Please let me know if that’s okay. Thanks!

**Comment by VedantMadane** [comment]
I'm fixing MobileNet v1 and v2 - they currently default to BILINEAR but should use BICUBIC to match the timm implementation. Will submit a PR shortly.

**Comment by sailaxmitumu2000** [comment]
Hi @NielsRogge, I'd like to work on fixing the default interpolation for the ResNet image processor by verifying it against the original torchvision/timm implementation and adding a pixel-value assertion test similar to existing vision models.

**Comment by Rocketknight1** [comment]
Closing this because it's attracting way too many code agents, and they're not handling it well!
